query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -253,47 +255,125 @@ public QueryResult selectUnique(String eventCollection, String targetProperty, T
      * This is the most flexible way to run a query. Use {@link Builder} to
      * build all the query arguments to run the query.
      *
-     * @param request     The {@link KeenQueryRequest} to be executed.
+     * @param request The {@link KeenQueryRequest} to be executed.
      * @return The {@link QueryResult} result.
      * @throws IOException If there was an error communicating with the server or
      * an error message received from the server.
      */
     public QueryResult execute(KeenQueryRequest request) throws IOException {
-        Map<String, Object> queryArgs = request.constructRequestArgs();
-        URL url = request.getRequestURL(this.requestUrlBuilder, this.project.getProjectId());
+        Map<String, Object> response = getMapResponse(request);
 
-        Map<String, Object> postResponse = postRequest(project, url, queryArgs);
+        return rawMapResponseToQueryResult(request, response);
+    }
 
-        QueryResult result = null;
-        boolean isFunnel = request instanceof Funnel;
+    /**
+     * Provides an implementation of {@link SavedQueries} for performing operations against
+     * the <a href="https://keen.io/docs/api/#saved-queries">Saved/Cached Query API</a> endpoints.
+     *
+     * @return The SavedQueries implementation.
+     */
+    public SavedQueries getSavedQueriesInterface() {
+        return new SavedQueriesImpl(this);
+    }
+
+    Map<String, Object> getMapResponse(KeenQueryRequest request) throws IOException {
+        // Throw are runtime if the return type isn't as expected.
+        // TODO : Throw a more specific exception
+        @SuppressWarnings("unchecked")
+        Map<String, Object> response = getResponse(Map.class, request);
+
+        return response;
+    }
+
+    List<Object> getListResponse(KeenQueryRequest request) throws IOException {
+        // Throw are runtime if the return type isn't as expected.
+        // TODO : Throw a more specific exception
+        @SuppressWarnings("unchecked")
+        List<Object> response = getResponse(List.class, request);
+
+        return response;
+    }
+
+    private <T> T getResponse(Class<T> containerType, KeenQueryRequest request) throws IOException {
+        Object response = getResponse(request);
+
+        if (!containerType.isAssignableFrom(response.getClass())) {
+            throw new RuntimeException(); // TODO : More specific
+        }
+
+        return containerType.cast(response);
+    }
+
+    private QueryResult rawMapResponseToQueryResult(KeenQueryRequest request,
+                                                    Map<String, Object> response) {
+        // TODO : Eventually moving result parsing out of this class might be a good plan, like
```

Comment:
Probably should just create an issue for this one, since it's going to be a large refactor.